Im folgenden liegt uns eine Bücherliste names bookstore als csv-Datei vor. Diese beinhaltet gebrauchte Bücher mit entsprechend dazu kennzeichnenden Eigenschaften.

Es gibt viele Möglichkeiten, wie wir uns numerische Spalten anschauen können. Eine Möglichkeit hierbei ist bspw. die Betrachtung als kategorische Spalte. Es kann in vielen Fällen genau so von Interesse sein bspw. welches der häufigste Verkaufspreis oder das beliebteste Publikationsjahr ist.

Beantworte folgende Fragen:

In [1]:
import pandas as pd

df = pd.read_csv("bookstore.csv")

df.head()

Title                       Author  \
0             The Lord of the Rings              Tolkien, J.R.R.   
1                         The Bible  International Bible Society   
2          The Pillars of the Earth                 Follett, Ken   
3  Perfume: The Story of a Murderer             Süskind, Patrick   
4                 The Little Prince    Saint-Exupéry, Antoine de   

   Publication Year  List Price  Sale Price  Condition  
0              1992        7.99        2.69   like new  
1              2003        7.50        2.49   like new  
2              2002       19.95        1.49  very good  
3              2003       19.95        1.29  very good  
4              2006       24.95        2.59       good

Was ist das Jahr mit den häufigsten Publikationen?

Aufgabe 1:

Um das Jahr mit den meisten Publikationen herauszufinden, bietet sich die value_counts() Methode an, vor allem weil diese die Ausgabe bereits sortiert. Hier können wir direkt vom ersten Eintrag ablesen, dass im Jahr 2003 mit einer Anzahl von 6 die meisten Publikationen waren. Wenn man möchte kann man die Ausgabe durch die iloc[,] Methode auf die ersten Zeilen reduzieren. Hier braucht man nur den ersten Parameter, da man hier nur bestimmte Zeilen auswählen möchte.

In [2]:
# häufigste Publikation
display(df.loc[:, "Publication Year"].value_counts().iloc[:3])

Publication Year
2003    6
2006    5
2005    3
Name: count, dtype: int64

Gibt es mehr unterschiedliche Listen- oder Verkaufspreise?

Aufgabe 2:

Die Anzahl einzigartiger Einträge findet man über die nunique() Methode heraus. Diese kann man nach der Ausgabe vergleichen und stellt fest, dass es 3 Listenpreise mehr als Verkaufspreise gibt.

In [3]:
print("Anzahl einzigartiger Listenpreise:   ", df.loc[:, "List Price"].nunique())
print("Anzahl einzigartiger Verkaufspreise: ", df.loc[:, "Sale Price"].nunique())

Anzahl einzigartiger Listenpreise:    11
Anzahl einzigartiger Verkaufspreise:  8


Vergleiche den häufigsten Listenpreis mit dem häufigsten Verkaufspreis.

Aufgabe 3:
Genau wie bei Aufgabe 1 kann man hier die value_counts() Methode benutzen. Hierdurch kann man feststellen, dass der häufigste Listenpreis 8 mal 7.99 und der häufigste Verkaufspreis 15 mal 1.29 ist. D.h. der häufigste Verkaufspreis kommt fast doppelt so oft vor, wie der häufigste Listenpreis. Bei nur 30 Zeilen heißt dass, dass die Hälfte der Bücher für 1.29 verkauft werden.

In [4]:
# Häufigster Listen- und Verkaufspreis 
display(df.loc[:, "List Price"].value_counts().iloc[:3])
display(df.loc[:, "Sale Price"].value_counts().iloc[:3])

List Price
7.99    8
6.99    6
7.50    3
Name: count, dtype: int64

Sale Price
1.29    15
1.49     4
2.49     3
Name: count, dtype: int64